sources/packages/web/src/views/student/StudentApplicationSummary.vue

```diff
@@ -38,14 +58,85 @@ export default {
     StudentApplications,
     CheckValidSINBanner,
     HeaderNavigator,
+    ConfirmEditApplication,
+    CancelApplication,
   },
   setup() {
     const hasRestriction = ref(false);
     const restrictionMessage = ref("");
+    const programYear = ref({} as ProgramYearOfApplicationDto);
+    const router = useRouter();
+    const toast = useToastMessage();
+    const editApplicationModal = ref({} as ModalDialog<boolean>);
+    const showModal = ref(false);
+    const selectedApplicationId = ref(0);
+    const reloadData = ref(false);
+
+    const openConfirmCancel = (id: number) => {
+      showModal.value = true;
+      selectedApplicationId.value = id;
+    };
+
+    const showHideCancelApplication = () => {
+      showModal.value = !showModal.value;
+    };
+
+    const getProgramYear = async (applicationId: number) => {
+      programYear.value = await ApplicationService.shared.getProgramYearOfApplication(
```

Comment:
The method can return the program year id and avoid creating a unnecessary local variable.
For this scenario, I don't think that we really need a method to do it, _getProgramYearOfApplication_ can be called directly inside the _editApplication_.
Going further, not part of this PR, I believe that we should remove the need of having the program year in order to call application.

Reply:
Andrew, `editApplication` is actually doing go to Edit Application page(maybe I will rename the function to avoid confusion) and from this API 
![image](https://user-images.githubusercontent.com/77353155/155233095-76dbd3a5-6c49-4728-86d0-96340caac231.png)
its getting formname and programyear id
![image](https://user-images.githubusercontent.com/77353155/155232728-bb36b40f-5323-454c-986c-d62a4ed3a0ba.png)
. this variable is confusing, i will rename the variable

Reply:
but I got the point here whenever there is only need to id, we can create an API only to return ID